### DISCUSSION TITLE
Recording retention. Is this expected behaviour?

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#3562] [Ask A Question]
I'm not sure if this is a bug or expected behaviour

I have the following in the config.yml

    record:
    enabled: True
    expire_interval: 60
    events:
        retain:
            default: 14
            mode: all
            objects:
                person: 10
                cat: 2
                dog: 2
                car: 7

I don't do continuous recording and the only objects I track are the four listed above but I notice that while the entries for each category disappear off the event list after their respective number of days the actual video is retained on the disk for the full 14 days. 

Is this how it should be?

**Comment by NickM-27** [comment]
Based on the code here https://github.com/blakeblackshear/frigate/blob/08f573aaa5ec0b9bf3d4b40388942254448b902a/frigate/events.py#L186 this seems to make sense but once the event is gone the recording files should be cleaned up on the next recording cleanup. Will look more into it and see.

**Comment by rea-ducks** [comment]
Hi,

I'm having a retention problem too, maybe same issue?
I've just noticed, that my frigate ran out of disk space because it still has recordings lingering from 2022-07-27 on though it's not supposed to.
I'm using frigate since about a month and the first 14 days it worked like a charm, just kept recordings for two days. Now I see hundrets of megabytes each day since above date. Having an event in progress since than would explain it to me, but I couldn't find one.
Here's my config running 0.10:

```
snapshots:
  enabled: True
  timestamp: True
  retain:
    default: 3
record:
  enabled: True
  retain:
    days: 0
    mode: active_objects
  events:
    retain:
      default: 2
      mode: active_objects

**Reply by NickM-27** [reply]
Need to see the entire config to get a full understanding of what's going on in your case

**Reply by rea-ducks** [reply]
Okay, here it comes:
```
mqtt:
  host: 192.168.1.2
  user: username
  password: password
ffmpeg:
  global_args: -hide_banner -loglevel warning
  hwaccel_args: []
  input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1
  Kamera:
    ffmpeg:
      inputs:
        - path: rtsp://username:password@192.168.0.0:554/stream=1
          roles:
            - detect
            - rtmp
          input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1
        - path: rtsp://username:password@192.168.1.100:554/stream=0
          roles:
            - record
    rtmp:
      enabled: True
    detect:
      enabled: True
      width: 448
      height: 800
      stationary:
        max_frames:
          objects:
            person: 100
    objects:
      track:
        - person
        - car
      filters:
        car:
          mask:
            - 0,21,0,97,145,91,146,21
snapshots:
  enabled: True
  timestamp: True
  retain:
    default: 3
timestamp_style:
  position: "tl"
  format: "%Y-%m-%d %H:%M:%S"
record:
  enabled: True
  retain:
    days: 0
    mode: active_objects
  events:
    retain:
      default: 2
      mode: active_objects
database:
  path: /db/frigate.db

**Reply by blakeblackshear** [reply]
Are these old recordings visible in the recordings UI? Can you watch them? If so, can you post your JSON config from the debug page?
